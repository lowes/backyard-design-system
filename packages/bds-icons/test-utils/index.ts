import { customRender } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }
