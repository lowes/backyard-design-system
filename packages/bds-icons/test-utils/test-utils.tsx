import * as React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import * as theme from '@lowes-tech/bds-tokens/v2/light/theme'

/**
 * Render Wrapper Helper
 *
 * This component wraps all test renders from react-testing-library
 * Uses default ThemeProvider with light
 */
const RenderWrapper = ({ children }) => {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

/**
 * Function to override custom render with
 *
 * @param {*} ui
 * @param {*} options
 */
const customRender = (ui, options) => render(ui, { wrapper: RenderWrapper, ...options })

export { customRender }
