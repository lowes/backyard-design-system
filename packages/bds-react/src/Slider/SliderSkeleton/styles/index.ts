import styled from 'styled-components'

export const HorizontalWrapper = styled.div`
    display: flex;
    flex-direction: row;
    align-items: center;

    .skeleton--slider-bar {
        margin-left: ${({ theme }) => theme.sizes.size_16};
        margin-right: ${({ theme }) => theme.sizes.size_16};
    }

    .skeleton--slider-input {
        margin-left: ${({ theme }) => theme.sizes.size_16};
    }
`

export const VerticalWrapper = styled.div`
    display: flex;
    flex-direction: row;

    .skeleton--slider-values {
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        margin-left: ${({ theme }) => theme.sizes.size_12};
    }
`