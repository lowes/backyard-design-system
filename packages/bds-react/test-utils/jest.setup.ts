import * as React from 'react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

jest.setTimeout(1000 * 60 * 20)

// Mock `useRandomId` hook
// gives consistent ids instead
// of randomly generated ones for
// snapshot tests
let mockId = 0

jest.mock('../src/utils/hooks/useRandomId', () => {
    return jest.fn((id) => id || `${mockId++}`)
})

// Before each test...
beforeEach(() => {
    // Reset `useRandomId` counter
    mockId = 0
})
