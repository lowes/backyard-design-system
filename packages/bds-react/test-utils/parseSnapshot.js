const parseSnapshot = (content) => {
    const data = get_data(content)

    return Object.keys(data).reduce(
        (parsed, key) => ({ ...parsed, [key]: parse(data[key]) }),
        {},
    )
}

function parse(value) {
    const reverted_value = revert_extra_line_breaks(value).replace(/\n/g, '\\n')

    try {
        return reverted_value === 'undefined'
            ? undefined
            : JSON.parse(reverted_value)
    } catch (error) {
        // istanbul ignore next
        throw new Error(
            `Unexpected error while parsing:\n\n>>> start >>>\n${reverted_value}\n<<< end <<<\n\n${
            error.message
            }`,
        )
    }
}

// https://github.com/facebook/jest/blob/v20.0.4/packages/jest-snapshot/src/utils.js#L128
function revert_extra_line_breaks(str) {
    return str.includes('\n') ? str.slice(1, -1) : str
}

// https://github.com/facebook/jest/blob/v20.0.4/packages/jest-snapshot/src/utils.js#L98
function get_data(content) {
    const data = Object.create(null)
    const populate = new Function('exports', content)
    
    populate(data)
    
    return data
}

export {
    parseSnapshot
}

export default parseSnapshot
