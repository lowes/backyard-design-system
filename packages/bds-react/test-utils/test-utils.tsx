import * as React from 'react'
import { render } from '@testing-library/react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/ThemeProvider'

/**
 * Render Wrapper Helper
 *
 * This component wraps all test renders from react-testing-library
 * Uses default ThemeProvider with light
 */
const RenderWrapper = ({ children }) => {
    return <ThemeProvider theme="light">{children}</ThemeProvider>
}

/**
 * Function to override custom render with
 *
 * @param {*} ui
 * @param {*} options
 */
// @ts-ignore
const customRender = (ui, options = {}) => render(ui, { wrapper: RenderWrapper, ...options })

/**
 * Function to override custom mount with
 *
 * @param {*} ui
 * @param {*} options
 */
const customMount = (ui, options = {}) =>
    mount(ui, {
        wrappingComponent: ThemeProvider,
        wrappingComponentProps: { theme: 'light' },
        ...options,
    })

export { customRender, customMount }
