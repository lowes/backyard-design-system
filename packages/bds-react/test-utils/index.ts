// @ts-ignore
import { customRender, customMount } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }

// Override shallow method
export { customMount as mount }
