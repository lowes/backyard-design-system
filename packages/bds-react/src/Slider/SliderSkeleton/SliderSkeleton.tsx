import * as React from 'react'
import classnames from 'classnames'
import type { BDSForwardRef } from '../../utils/typings/BDSComponentProps'

import Skeleton from '../../Skeleton'
import { HorizontalWrapper, VerticalWrapper } from './styles'

/**
 * Backyards Slider skeleton component
 * 
 * Returns a series of Skeleton Rects in either a horizontal row or a
 * vertical stack depending on the orientation prop. Option prop for showing
 * a text input available only in the horizontal orientation.
 * 
 * <SliderSkeleton />
 */

const SliderSkeleton: BDSForwardRef<SliderSkeletonProps> = React.forwardRef<
    SliderSkeletonRef,
    SliderSkeletonProps
>(function SliderSkeleton (
    {
        orientation = 'horizontal',
        animate = false,
        showInput = false,
        className,
        ...props
    },
    ref,
) {
    let Component

    if (orientation === 'horizontal') {
        Component = () => (
            <HorizontalWrapper className={classnames(className, 'slider--skeleton', `orientation--${orientation}`)} {...props} ref={ref}>
                <Skeleton 
                    variant='rect'
                    height='size_24'
                    width ='size_16'
                    animate={animate}
                    className='skeleton--slider-value'
                />

                <Skeleton 
                    variant='rect'
                    height='size_20'
                    width='12.5rem'
                    animate={animate}
                    className='skeleton--slider-bar'
                />

                <Skeleton 
                    variant='rect'
                    height='size_24'
                    width ='size_16'
                    animate={animate}
                    className='skeleton--slider-value'
                />

                {
                    (showInput === true) ?
                        <Skeleton 
                            variant='rect'
                            height='3.375rem'
                            width='3.375rem'
                            animate={animate}
                            className='skeleton--slider-input'
                        /> : null
                }
            </HorizontalWrapper>
        )
    } else {
        Component = () => (
            <VerticalWrapper className={classnames(className, 'slider--skeleton', `orientation--${orientation}`)} {...props} ref={ref}>
                <Skeleton 
                    variant='rect'
                    height='12.5rem'
                    width='size_20'
                    animate={animate}
                    className='skeleton--slider-bar'
                />

                <div className='skeleton--slider-values'>
                    <Skeleton 
                        variant='rect'
                        height='size_24'
                        width ='size_16'
                        animate={animate}
                        className='skeleton--slider-value'
                    />

                    <Skeleton 
                        variant='rect'
                        height='size_24'
                        width ='size_16'
                        animate={animate}
                        className='skeleton--slider-value'
                    />
                </div>
            </VerticalWrapper>
        )
    }

    return (
        <Component />
    )
})

type SliderSkeletonRef = HTMLDivElement

interface SliderSkeletonProps {
    orientation?: string
    animate?: boolean
    showInput?: boolean
    className?: string
}

SliderSkeleton.bdsName = 'SliderSkeleton'

export { SliderSkeleton }

export type { SliderSkeletonProps, SliderSkeletonRef }

export default SliderSkeleton