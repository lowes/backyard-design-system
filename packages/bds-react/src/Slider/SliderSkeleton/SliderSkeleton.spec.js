import * as React from 'react'
import { render } from '../../../test-utils'
import SliderSkeleton from './SliderSkeleton'

describe('SliderSkeleton Snapshots', () => {
    test('SliderSkeleton default snapshot', () => {
        const { asFragment } = render(
            <SliderSkeleton />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('SliderSkeleton with input option', () => {
        const { asFragment } = render(
            <SliderSkeleton showInput />
        )

        expect(asFragment()).toMatchSnapshot()
    })


    test('SliderSkeleton animate snapshot', () => {
        const { asFragment } = render(
            <SliderSkeleton animate={true} />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('SliderSkeleton vertical orientation', () => {
        const { asFragment } = render(
            <SliderSkeleton orientation='vertical' />
        )

        expect(asFragment()).toMatchSnapshot()
    })
})