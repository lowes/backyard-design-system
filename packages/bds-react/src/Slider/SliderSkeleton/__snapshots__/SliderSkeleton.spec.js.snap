// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderSkeleton Snapshots SliderSkeleton animate snapshot 1`] = `
<DocumentFragment>
  .c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 1rem;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 1rem;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c2 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c2.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c2.variant--circle {
  width: 12.5rem;
  height: 1.25rem;
  border-radius: var(--bds-border-radius-circle);
}

.c2.variant--rect {
  width: 12.5rem;
  height: 1.25rem;
}

.c2.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c2.variant--rect.shape--squared {
  border-radius: 0;
}

.c2.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c2.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c2.variant--text.shape--squared {
  border-radius: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .skeleton--slider-bar {
  margin-left: 1rem;
  margin-right: 1rem;
}

.c0 .skeleton--slider-input {
  margin-left: 1rem;
}

<div
    class="c0 slider--skeleton orientation--horizontal"
  >
    <div
      class="c1 variant--rect shape--rounded skeleton--slider-value style--animated"
      height="size_24"
      width="size_16"
    />
    <div
      class="c2 variant--rect shape--rounded skeleton--slider-bar style--animated"
      height="size_20"
      width="12.5rem"
    />
    <div
      class="c1 variant--rect shape--rounded skeleton--slider-value style--animated"
      height="size_24"
      width="size_16"
    />
  </div>
</DocumentFragment>
`;

exports[`SliderSkeleton Snapshots SliderSkeleton default snapshot 1`] = `
<DocumentFragment>
  .c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 1rem;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 1rem;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c2 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c2.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c2.variant--circle {
  width: 12.5rem;
  height: 1.25rem;
  border-radius: var(--bds-border-radius-circle);
}

.c2.variant--rect {
  width: 12.5rem;
  height: 1.25rem;
}

.c2.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c2.variant--rect.shape--squared {
  border-radius: 0;
}

.c2.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c2.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c2.variant--text.shape--squared {
  border-radius: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .skeleton--slider-bar {
  margin-left: 1rem;
  margin-right: 1rem;
}

.c0 .skeleton--slider-input {
  margin-left: 1rem;
}

<div
    class="c0 slider--skeleton orientation--horizontal"
  >
    <div
      class="c1 variant--rect shape--rounded skeleton--slider-value style--default"
      height="size_24"
      width="size_16"
    />
    <div
      class="c2 variant--rect shape--rounded skeleton--slider-bar style--default"
      height="size_20"
      width="12.5rem"
    />
    <div
      class="c1 variant--rect shape--rounded skeleton--slider-value style--default"
      height="size_24"
      width="size_16"
    />
  </div>
</DocumentFragment>
`;

exports[`SliderSkeleton Snapshots SliderSkeleton vertical orientation 1`] = `
<DocumentFragment>
  .c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 1.25rem;
  height: 12.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 1.25rem;
  height: 12.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c2 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c2.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c2.variant--circle {
  width: 1rem;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c2.variant--rect {
  width: 1rem;
  height: 1.5rem;
}

.c2.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c2.variant--rect.shape--squared {
  border-radius: 0;
}

.c2.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c2.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c2.variant--text.shape--squared {
  border-radius: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 .skeleton--slider-values {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-left: 0.75rem;
}

<div
    class="c0 slider--skeleton orientation--vertical"
  >
    <div
      class="c1 variant--rect shape--rounded skeleton--slider-bar style--default"
      height="12.5rem"
      width="size_20"
    />
    <div
      class="skeleton--slider-values"
    >
      <div
        class="c2 variant--rect shape--rounded skeleton--slider-value style--default"
        height="size_24"
        width="size_16"
      />
      <div
        class="c2 variant--rect shape--rounded skeleton--slider-value style--default"
        height="size_24"
        width="size_16"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SliderSkeleton Snapshots SliderSkeleton with input option 1`] = `
<DocumentFragment>
  .c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 1rem;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 1rem;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c2 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c2.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c2.variant--circle {
  width: 12.5rem;
  height: 1.25rem;
  border-radius: var(--bds-border-radius-circle);
}

.c2.variant--rect {
  width: 12.5rem;
  height: 1.25rem;
}

.c2.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c2.variant--rect.shape--squared {
  border-radius: 0;
}

.c2.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c2.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c2.variant--text.shape--squared {
  border-radius: 0;
}

.c3 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c3.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c3.variant--circle {
  width: 3.375rem;
  height: 3.375rem;
  border-radius: var(--bds-border-radius-circle);
}

.c3.variant--rect {
  width: 3.375rem;
  height: 3.375rem;
}

.c3.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c3.variant--rect.shape--squared {
  border-radius: 0;
}

.c3.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c3.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c3.variant--text.shape--squared {
  border-radius: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .skeleton--slider-bar {
  margin-left: 1rem;
  margin-right: 1rem;
}

.c0 .skeleton--slider-input {
  margin-left: 1rem;
}

<div
    class="c0 slider--skeleton orientation--horizontal"
  >
    <div
      class="c1 variant--rect shape--rounded skeleton--slider-value style--default"
      height="size_24"
      width="size_16"
    />
    <div
      class="c2 variant--rect shape--rounded skeleton--slider-bar style--default"
      height="size_20"
      width="12.5rem"
    />
    <div
      class="c1 variant--rect shape--rounded skeleton--slider-value style--default"
      height="size_24"
      width="size_16"
    />
    <div
      class="c3 variant--rect shape--rounded skeleton--slider-input style--default"
      height="3.375rem"
      width="3.375rem"
    />
  </div>
</DocumentFragment>
`;
